core/dbt/context/common.py

```diff
@@ -398,8 +398,6 @@ def generate_base(model, model_dict, config, manifest, source_config,
         "exceptions": dbt.exceptions.wrapped_exports(model),
         "execute": provider.execute,
         "flags": dbt.flags,
-        # TODO: Do we have to leave this in?
```

Comment:
I'm a-ok with this. We've never had this documented and have actively discouraged users from touching it when they've asked.

Are you opposed to leaving the `manifest` in the context but continuing to not document / recommend using it? I know of exactly one project which is using `graph` to cleverly union together `source`s (I helped with the implementation 😉).

Longer term, the right answer would be to use the dynamic graph selection functionality described in #1212, but just leaving in `manifest` at least provides some sort of upgrade path for users that have poked around in the dbt codebase.

Curious to hear what you think!

Reply:
Ok, I've come up with a workaround to this that I think will be satisfactory to everyone and will be backwards-compatible with the one user we know of. It will break anyone using `graph.macros`, but `graph.macros` shouldn't be useful to anyone.